specs/online_order_spec.rb

```diff
@@ -1,90 +1,213 @@
+  before do
+    @normal_products = { "sandwich" => 9.99, "gum" => 1.0 }
+    @normal_order_id = 121
+    @customer_id = 21
+    @normal_order = Grocery::OnlineOrder.new(@normal_order_id, @normal_products,
+      @customer_id, :pending)
+    @normal_total = ((9.99 + 1.0) + ((9.99 + 1.0) * 0.075)).round(2) + 10.0
+  end
 
-# TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
-# You may also need to require other classes here
-
-# Because an OnlineOrder is a kind of Order, and we've
-# already tested a bunch of functionality on Order,
-# we effectively get all that testing for free! Here we'll
-# only test things that are different.
-
-xdescribe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
-      # Check that an OnlineOrder is in fact a kind of Order
-
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      @normal_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+      @normal_order.must_respond_to :customer
+      @normal_order.customer.must_be_kind_of Grocery::Customer
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      @normal_order.fulfillment_status.must_be_kind_of Symbol
+      @normal_order.fulfillment_status.must_equal :pending
+    end
+
+    it "Can provide its own the online order status" do
+      status = :shipped
+      test_order = Grocery::OnlineOrder.new(@normal_order_id,
+        @normal_products, @customer_id, status)
+
+      test_order.fulfillment_status.must_be_kind_of Symbol
+      test_order.fulfillment_status.must_equal status
+    end
+
+    it "Sets invalid status to pending" do
+      status = "ferrets"
+      test_order = Grocery::OnlineOrder.new(9999, {"gum" => 1.0 }, 21, status)
+      test_order.fulfillment_status.must_equal :pending
+    end
+
+    it "Does not need to be provided status" do
+      test_order = Grocery::OnlineOrder.new(9999, {"gum" => 1.0 }, 21)
+      test_order.fulfillment_status.must_equal :pending
     end
+
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      @normal_order.total.must_be_kind_of Float
+      @normal_order.total.must_equal @normal_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      test_order = Grocery::OnlineOrder.new(@normal_order_id, {}, 21, :pending)
+      test_order.total.must_be_kind_of Float
+      test_order.total.must_equal 0.0
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      test_order = Grocery::OnlineOrder.new(@normal_order_id, @normal_products,
+        @customer_id, :shipped)
+
+      assert_raises{test_order.add_product("apple", 1.0)}
```

Comment:
Similar to another note, it's better to use `must_raise` since we're doing spec-style testing.